File: FleemanWithNET/Models/BookingConfirmationDetails.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fleeman_with_dot_net.Models
{
    public class BookingConfirmationDetails
    {
        [Key]
        public int BookingConfirmationDetailsId { get; set; }
        public int BookingId { get; set; }


        [ForeignKey(nameof(BookingId))]
        public virtual BookingDetails Booking { get; set; }

        public int Car_Id { get; set; }

        [ForeignKey(nameof(Car_Id))]
        public virtual CarDetails Car { get; set; }
    }
}

File: FleemanWithJava/src/main/java/com/example/models/BookingConfirmationDetails.java
package com.example.models;

import jakarta.persistence.*;

@Entity
@Table(name = "Booking_Confirmation_Details")
public class BookingConfirmationDetails {

    @Id
    @Column(name = "booking_id")
    private Long bookingId; // still the primary key

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "booking_id", referencedColumnName = "bookingId", insertable = false, updatable = false)
    private BookingDetails booking; // this is your object

    @OneToOne
    @JoinColumn(name = "car_id", nullable = false)
    private CarDetails car;

    // Getters & Setters
    public Long getBookingId() {
        return bookingId;
    }
    public void setBookingId(Long bookingId) {
        this.bookingId = bookingId;
    }

    public BookingDetails getBooking() {
        return booking;
    }
    public void setBooking(BookingDetails booking) {
        this.booking = booking;
    }

    public CarDetails getCar() {
        return car;
    }
    public void setCar(CarDetails car) {
        this.car = car;
    }
}
